src/Common/Identifiers.hs

```diff
@@ -119,6 +119,7 @@ newtype FfiId = FfiId Identifier
 -- | ToIdentifier for user-defined variable, e.g., @x@
 newtype VarId = VarId Identifier
   deriving Eq
+  deriving Ord via Identifier
```

Comment:
While you're at it, can you add this to all other identifier types? They are likely candidates for being keys of maps.